executor/insert_common.go

```diff
@@ -847,30 +847,50 @@ func (e *InsertValues) adjustAutoRandomDatum(ctx context.Context, d types.Datum,
 		return d, nil
 	}
 
-	if !hasValue || d.IsNull() {
-		_, err := e.ctx.Txn(true)
```

Comment:
better keep this, cause `allocAutoRandomID` will use `txn.startTs` to shard.

Reply:
Txn is moved to line 876.

Reply:
nice